modules/statistics-provider-influx/src/main/java/org/opencastproject/statistics/provider/influx/provider/InfluxRunningTotalStatisticsProvider.java

```diff
@@ -0,0 +1,189 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.statistics.provider.influx.provider;
+
+import org.opencastproject.statistics.api.DataResolution;
+import org.opencastproject.statistics.api.ResourceType;
+import org.opencastproject.statistics.api.TimeSeries;
+import org.opencastproject.statistics.api.TimeSeriesProvider;
+import org.opencastproject.statistics.provider.influx.StatisticsProviderInfluxService;
+import org.opencastproject.util.data.Tuple;
+
+import org.influxdb.dto.BoundParameterQuery;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class InfluxRunningTotalStatisticsProvider extends InfluxStatisticsProvider implements TimeSeriesProvider {
+  private final Set<InfluxProviderConfiguration.InfluxProviderSource> sources;
+
+  public InfluxRunningTotalStatisticsProvider(
+          StatisticsProviderInfluxService service,
+          String id,
+          ResourceType resourceType,
+          String title,
+          String description,
+          Set<InfluxProviderConfiguration.InfluxProviderSource> sources) {
+    super(service, id, resourceType, title, description);
+    this.sources = sources;
+  }
+
+  private static double reduceResult(double previousTotal, QueryResult.Result newResult) {
+    if (newResult.getSeries() == null) {
+      return previousTotal;
+    }
+    return previousTotal + newResult
+            .getSeries()
+            .stream()
+            .reduce(0.0, InfluxRunningTotalStatisticsProvider::reduceSeries, Double::sum);
+  }
+
+  private static double reduceSeries(double previousSeriesTotal, QueryResult.Series newSeries) {
+    if (newSeries.getValues().isEmpty()) {
+      return previousSeriesTotal;
+    }
+    if (newSeries.getValues().size() > 1) {
+      throw new RuntimeException("invalid results returned for aggregation");
+    }
+    final List<Object> objects = newSeries.getValues().get(0);
+    if (objects.size() != 2) {
+      throw new RuntimeException("invalid results returned for aggregation");
+    }
+    final Object o = objects.get(1);
+    if (o == null) {
+      return previousSeriesTotal;
+    }
+    if (!(o instanceof Double)) {
+      throw new RuntimeException("invalid results returned for aggregation");
+    }
+    return previousSeriesTotal + (Double) o;
+  }
+
+  private double getPreviousTotal(
+          final InfluxProviderConfiguration.InfluxProviderSource source, final String resourceId, final Instant from) {
+    final Query beforeQuery = BoundParameterQuery.QueryBuilder
+            .newQuery("SELECT SUM(" + source.getAggregationVariable() + ") FROM " + source.getMeasurement() + " WHERE "
```

Comment:
Why did you parameterize some of these variables, and not others?

Reply:
More than one reason:

1. The parametrized values are user generated ones, preventing injections.
2. It's copy&pasted from the `TimeSeriesInfluxDbProvider` (so we'd have to change it there, risking merge conflicts.
3. I don't even know if Influx allows arbitrary placeholders.

Reply:
Fair enough!